votingpool/db.go

```diff
@@ -56,13 +59,79 @@ type dbSeriesRow struct {
 	privKeysEncrypted [][]byte
 }
 
+// getUsedAddrBucketID returns the used addresses bucket ID for the given series
+// and branch. It has the form seriesID:branch.
+func getUsedAddrBucketID(seriesID uint32, branch Branch) []byte {
+	return bytes.Join(
+		[][]byte{uint32ToBytes(seriesID), uint32ToBytes(uint32(branch))}, []byte(":"))
```

Comment:
This is creating a fair amount of garbage (2 byte slices for the seriesID and branch, a bytes slice for the colon, and a slice of slices for the fields).  It would obviously depend on how often this is called to determine whether it's worth it, but it would be more efficient to do something like:

```Go
var bucketID [9]
binary.LittleEndian.PutUint32(bucketID[0:4], seriesID)
bucketID[4] = ':'
binary.LittleEndian.PutUint32(bucketID[5:9], uint32(branch))
return bucketID[:]
```